lua/telescope/pickers.lua

```diff
@@ -419,11 +419,11 @@ function Picker:find()
     pcall(a.nvim_buf_set_option, prompt_bufnr, "filetype", "TelescopePrompt")
     pcall(a.nvim_buf_set_option, results_bufnr, "filetype", "TelescopeResults")
 
-    -- TODO(async): I wonder if this should actually happen _before_ we nvim_buf_attach.
-    -- This way the buffer would always start with what we think it should when we start the loop.
     if self.initial_mode == "insert" or self.initial_mode == "normal" then
       -- required for set_prompt to work adequately
-      vim.cmd [[startinsert!]]
+      vim.schedule(function()
```

Comment:
```diff
diff --git a/lua/telescope/pickers.lua b/lua/telescope/pickers.lua
index d91ae5c..3889ebf 100644
--- a/lua/telescope/pickers.lua
+++ b/lua/telescope/pickers.lua
@@ -422,6 +422,10 @@ function Picker:find()
     if self.initial_mode == "insert" or self.initial_mode == "normal" then
       -- required for set_prompt to work adequately
       vim.schedule(function()
+        if vim.fn.mode() == 'v' then
+          local key = vim.api.nvim_replace_termcodes("<ESC>", true, false, true)
+          vim.api.nvim_feedkeys(key, "i", true)
+        end
         vim.cmd [[startinsert!]]
       end)
       if self.default_text then
```
Maybe it's not the best solution but it lets me avoid manual exiting from _insert (visual)_ mode.

Reply:
I think we should deal with opening telescope from visual mode some other time. Ill open a new issue

Reply:
Better idea, we make your issue about problems opening telescope from visual mode

Reply:
@przepompownia Its fixed because i already had to fix opening telescope from insert mode, because previously that was working. So yeah why not

Reply:
Commented outside this thread: https://github.com/nvim-telescope/telescope.nvim/pull/1600#issuecomment-1106402968